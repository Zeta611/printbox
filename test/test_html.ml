let b =
  let open PrintBox in
  tree
    (frame @@ text "root")
    [
      link ~uri:"#HiddenAnchor" @@ text "link to a hidden anchor";
      vlist ~bars:true [ text "1"; text "2"; text "3"; text "4"; text "5" ];
      frame @@ text "child 1";
      text "child 2";
      frame
      @@ tree empty
           [ tree (frame @@ text "header 3") [ frame @@ text "subchild 3" ] ];
      tree empty [ tree (frame @@ text "header 4") [ text "subchild 4" ] ];
      frame @@ tree (text "header 5") [ text "subchild 5" ];
      frame @@ text "child 5";
      text "separator";
      tree
        (hlist ~bars:false [ text "entry 0.1"; text "entry 0.2" ])
        [ text "child 5.5" ];
      text "separator";
      tree
        (hlist ~bars:false [ text "entry 1"; frame @@ text "entry 2" ])
        [ text "child 6" ];
      anchor ~id:"VisibleAnchor" @@ text "anchor (visible)";
      tree
        (hlist ~bars:true
           [
             anchor ~id:"HiddenAnchor" empty;
             text "entry 3";
             frame @@ text "entry 4";
           ])
        [ text "child 7" ];
      text "separator after hidden anchor";
      anchor ~id:"HiddenAnchor2" empty;
      tree
        (vlist ~bars:false [ text "entry 5"; frame @@ text "entry 6" ])
        [ text "child 8" ];
      text "separator";
      tree
        (vlist ~bars:true [ text "entry 7"; frame @@ text "entry 8" ])
        [ text "child 9" ];
    ]

let () =
  print_endline
  @@ PrintBox_html.(to_string ~config:Config.(tree_summary true default)) b
