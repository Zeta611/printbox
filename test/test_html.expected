<div><details><summary><span style="border:thin solid"><span>root</span></span></summary><ul><li><div><a href="#HiddenAnchor"><div>link to a hidden anchor</div></a></div></li><li><table class="framed"><tr><td><div>1</div></td></tr><tr><td><div>2</div></td></tr><tr><td><div>3</div></td></tr><tr><td><div>4</div></td></tr><tr><td><div>5</div></td></tr></table></li><li><div style="border:thin solid"><div>child 1</div></div></li><li><div>child 2</div></li><li><div style="border:thin solid"><div><div></div><ul><li><details><summary><span style="border:thin solid"><span>header 3</span></span></summary><ul><li><div style="border:thin solid"><div>subchild 3</div></div></li></ul></details></li></ul></div></div></li><li><div><div></div><ul><li><details><summary><span style="border:thin solid"><span>header 4</span></span></summary><ul><li><div>subchild 4</div></li></ul></details></li></ul></div></li><li><div style="border:thin solid"><details><summary><span>header 5</span></summary><ul><li><div>subchild 5</div></li></ul></details></div></li><li><div style="border:thin solid"><div>child 5</div></div></li><li><div>separator</div></li><li><details><summary><span><span><span><span>entry 0.1</span></span>&nbsp;<span><span>entry 0.2</span></span></span></span></summary><ul><li><div>child 5.5</div></li></ul></details></li><li><div>separator</div></li><li><details><summary><span><span><span><span>entry 1</span></span>&nbsp;<span><span style="border:thin solid"><span>entry 2</span></span></span></span></span></summary><ul><li><div>child 6</div></li></ul></details></li><li><a id="VisibleAnchor" href="#VisibleAnchor"><div>anchor (visible)</div></a></li><li><details><summary><span><span style="border:thin dotted"><span style="border:thin dotted"><a id="HiddenAnchor"></a></span>&nbsp;<span style="border:thin dotted"><span>entry 3</span></span>&nbsp;<span style="border:thin dotted"><span style="border:thin solid"><span>entry 4</span></span></span></span></span></summary><ul><li><div>child 7</div></li></ul></details></li><li><div>separator after hidden anchor</div></li><li><details><summary><span><span><span><span>entry 5</span></span></span><br/><span><span><span style="border:thin solid"><span>entry 6</span></span></span></span></span></summary><ul><li><div>child 8</div></li></ul></details></li><li><div>separator</div></li><li><details><summary><span><span style="border:thin dotted"><span style="border:thin dotted"><span>entry 7</span></span></span><br/><span style="border:thin dotted"><span style="border:thin dotted"><span style="border:thin solid"><span>entry 8</span></span></span></span></span></summary><ul><li><div>child 9</div></li></ul></details></li></ul></details></div>

